integrationtests.py

```diff
@@ -477,6 +473,21 @@ def testHitsSimple(self):
             self.assertEqual(newPreprocessorCalls, oldPreprocessorCalls + i, str(cmd))
 
 
+class TestNoDirectCalls(unittest.TestCase):
+    def testPreprocessorFailure(self):
+        cache = clcache.ObjectCache()
+
+        oldStats = clcache.CacheStatistics(cache)
+
+        cmd = CLCACHE_CMD + ["/nologo", "/c", "doesnotexist.cpp"]
+        env = dict(os.environ, CLCACHE_NODIRECT=str("1"))
```

Comment:
Ah, is that so? Excellent! I mostly based this on the existing code, ahem - this is now my first 'real' Python 3 project, I still need to become more familiar with all the goodies. :-)